cypress/integration/core-opensearch-dashboards/opensearch-dashboards/apps/query_enhancements/saved_search.spec.js

```diff
@@ -0,0 +1,513 @@
+  },
+  PPL: {
+    ...allowedSearchOperations.PPL,
+    language: 'PPL',
+    dataset: INDEX_PATTERN_WITH_TIME,
+    queryString: `source = ${INDEX_PATTERN_WITH_TIME} | where bytes_transferred > 9950`,
+    hitCount: 101,
+    sampleTableData: [],
+    saveName: 'ppl-index-pattern-01',
+  },
+};
+const indexTestConfigurations = {
+  'OpenSearch SQL': {
+    ...allowedSearchOperations['OpenSearch SQL'],
+    language: 'OpenSearch SQL',
+    dataset: INDEX_WITH_TIME_1,
+    queryString: `SELECT * FROM ${INDEX_WITH_TIME_1} WHERE bytes_transferred > 9950`,
+    hitCount: undefined,
+    sampleTableData: [],
+    saveName: 'sql-index-01',
+  },
+  PPL: {
+    ...allowedSearchOperations.PPL,
+    language: 'PPL',
+    dataset: INDEX_WITH_TIME_1,
+    queryString: `source = ${INDEX_WITH_TIME_1} | where bytes_transferred > 9950`,
+    hitCount: 50,
+    sampleTableData: [],
+    saveName: 'ppl-index-01',
+  },
+};
+const allTestConfigurations = [
+  ...Object.values(indexPatternTestConfigurations),
+  ...Object.values(indexTestConfigurations),
+];
+
+// Maps a dataset that are used in this file to the exact string that dataset
+// corresponds to in the saved search API response
+const mapDatasetToType = (dataset) => {
+  switch (dataset) {
+    case INDEX_PATTERN_WITH_TIME:
+      return 'INDEX_PATTERN';
+    case INDEX_WITH_TIME_1:
+      return 'INDEXES';
+    default:
+      return 'unknown dataset';
+  }
+};
+
+// Maps a language that are used in this file to the exact string that the language
+// corresponds to in the saved search API response
+const mapLanguageToApiResponseString = (language) => {
+  switch (language) {
+    case 'DQL':
+      return 'kuery';
+    case 'Lucene':
+      return 'lucene';
+    case 'OpenSearch SQL':
+      return 'SQL';
+    case 'PPL':
+      return 'PPL';
+    default:
+      return 'unknown language';
+  }
+};
+
+// Escapes special characters for the editor
+const prepareQueryStringForEditor = (queryString) => {
+  return queryString.replaceAll(/([{}])/g, (char) => `{${char}}`);
+};
+
+// In theory this function is not needed as we should be able to use
+// the custom command navigateToWorkSpaceSpecificPage, but when using that
+// to go to the discover page, it sometimes leads to a blank page
+const navigateToDiscoverPage = () => {
+  cy.navigateToWorkSpaceHomePage(workspaceName);
+  cy.getElementByTestId('headerAppActionMenu').should('be.visible');
+};
+
+// Sets the dataset for the search. Since INDEXES are not saved to the dataset,
+// we need to click through various buttons to manually add them
+const setDataset = (dataset) => {
+  const datasetType = mapDatasetToType(dataset);
+  cy.getElementByTestId('datasetSelectorButton').should('be.visible').click();
+
+  if (datasetType === 'INDEX_PATTERN') {
+    cy.get(`[title="${dataset}"]`).click();
+  } else if (datasetType === 'INDEXES') {
+    cy.getElementByTestId('datasetSelectorAdvancedButton').click();
+    cy.contains('span', 'Indexes').click();
```

Comment:
Abby used the following for click indexes
```
      cy.getElementByTestId(`datasetSelectorButton`).click();
      cy.getElementByTestId(`datasetSelectorAdvancedButton`).click();
      cy.get(`[title="Indexes"]`).click();
      cy.get(`[title=${DATASOURCE_NAME}]`).click();
      cy.get(`[title="data_logs_small_time_1"]`).click(); // Updated to match loaded data
      cy.getElementByTestId('datasetSelectorNext').click();
```
I think we should put it in a utility for select index (index name) under specific datasource name.

Reply:
Created the utility and also refactored the other test that you are referring to, to use the utility